### PR TITLE
Move cert-exporter to app catalog

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/10177

Please note this component has been deployed to control planes by `architect`. This is no longer the case as it will be pushed to `aws/kvm/azure-app-collection`.

~~Also e2e basic test has been disabled.~~ Moving this to a separate PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -192,10 +97,44 @@ workflows:
             branches:
               only: master
 
-      - deploy:
+      - architect/push-to-app-collection:
+          context: architect
+          name: push-cert-exporter-to-aws-app-collection
+          app_name: "cert-exporter"
```
Comment (corest) [inline .circleci/config.yml:103]: I think all these jobs are missing `app_namespace` field set to monitoring. `cert-exporter` is running in monitoring namespace

File: error.go [hunk 2]
```diff
@@ -4,15 +4,11 @@ import (
...
-	Kind: "executionFailedError",
-}
-
 var invalidConfigError = &microerror.Error{
 	Kind: "invalidConfigError",
 }
 
 // IsInvalidConfig asserts invalidConfigError.
-func IsInvalidConfig(err error) bool {
+func IsInvalidConfig(err error) bool { // nolint: unused, deadcode
```
[the remark this answers is not in the document]
Comment (kopiczko) [reply error.go:12]: I guess only matcher is unused. Interesting it fails the check on exported function.

File: .circleci/config.yml (hunk 1 above)
Comment (kopiczko) [reply .circleci/config.yml:103]: To clarify:

```suggestion
          app_name: "cert-exporter"
          app_namespace: "monitoring"
```

**Comment by kopiczko** [comment]
Can we please separate removing integration test removal? I think `integration` directory is not removed and it was hard to catch.

File: error.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (wtty-fool) [reply error.go:12]: It's just the matcher that is usused. I can either remove the matcher alone (which is not very compliant with our `fmt`), or remove the 1 place `invalidConfigError` is used.

[the remark this answers is not in the document]
Comment (kopiczko) [reply error.go:12]: I'd say let's keep it. But I still don't get why linter considers exported funcs unused.

**Comment by wtty-fool** [comment]
Tested on _ghost_.

**Comment by kopiczko** [comment]
You need to restore cert-exporter-chart to make e2e work again.

**Pull request merged**